question/bank/exporttoxml/classes/exporttoxml_helper.php

```diff
@@ -0,0 +1,67 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Helper class for export to xml.
+ *
+ * @package    qbank_exporttoxml
+ * @copyright  2021 Catalyst IT Australia Pty Ltd
+ * @author     Safat Shahin <safatshahin@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace qbank_exporttoxml;
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * Class exporttoxml_helper
+ *
+ * @package    qbank_exporttoxml
+ * @copyright  2021 Catalyst IT Australia Pty Ltd
+ * @author     Safat Shahin <safatshahin@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class exporttoxml_helper {
+
+    /**
+     * Get the URL to export a single question (exportone.php).
+     *
+     * @param \stdClass|\question_definition $question the question definition as obtained from
+     *      question_bank::load_question_data() or question_bank::make_question().
+     *      (Only ->id and ->contextid are used.)
+     * @return \moodle_url the requested URL.
```

Comment:
can you add please the doc for throwing exception `* @throws \moodle_exception` as moodle_url throws that one.

Reply:
i was following the document from moodle and didnt use throws anywhere:
https://docs.moodle.org/dev/Setting_up_PhpStorm

which states:
Disable missing @throws tag warning in "Preferences / Project Settings / Inspections / PHP / PHPDoc"

i was assuming throws tag is something not recommended, any thoughts?

Reply:
Hi @safatshahin some recent core things that use the @throws tag have been added. But, if core documentation says that is not mandatory, then is fine. Thank you.